source/components/molecules/TimeSlotPicker/TimeSlotPicker.tsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { View, Text } from "react-native";
+import { ScrollView } from "react-native-gesture-handler";
+import stringify from "json-stable-stringify";
+import { DayPicker, TimeSpanButton } from "..";
+
+interface TimeSlotPickerProps {
+  value: any;
+  onChange: (newObject: any) => void;
+  availableTimes: any;
+}
+
+const TimeSlotPicker = ({
+  value,
+  onChange,
+  availableTimes,
+}: TimeSlotPickerProps): JSX.Element => {
+  const dates = Object.keys(availableTimes);
+
+  const currentDate = value && value.date ? value.date : "";
+  const timeObject = value && value.times ? value.times : [];
```

Comment:
what do we even mean with `timeObject`? And also, why do we default to an empty array when the constant name is ending with the word object?

Reply:
I messed up the types here 😅
I have now added types and this is no longer a problem.